recipes/xrandr.lua

```diff
@@ -0,0 +1,146 @@
+
+local awful     = require("awful")
+local naughty   = require("naughty")
+local beautiful = require("beautiful");
+
+-- A path to a fancy icon
+local icon_path = "";
+
+-- Get active outputs
+local function outputs()
+   local outputs = {}
+   local xrandr = io.popen("xrandr -q")
+
+   if xrandr then
+      for line in xrandr:lines() do
+         local output = line:match("^([%w-]+) connected ")
+         if output then
+            outputs[#outputs + 1] = output
+                                   end
+      end
+      xrandr:close()
+   end
+
+   return outputs
+end
+
+local function arrange(out)
+   -- We need to enumerate all the way to combinate output. We assume
+   -- we want only an horizontal layout.
+
+   local choices  = {}
+   local previous = { {} }
+   for i = 1, #out do
+      -- Find all permutation of length `i`: we take the permutation
+      -- of length `i-1` and for each of them, we create new
+      -- permutations by adding each output at the end of it if it is
+      -- not already present.
+      local new = {}
+      for _, p in pairs(previous) do
+         for _, o in pairs(out) do
+            if not awful.util.table.hasitem(p, o) then
+               new[#new + 1] = awful.util.table.join(p, {o})
+            end
+         end
+      end
+      choices = awful.util.table.join(choices, new)
+      previous = new
+   end
+
+   return choices
+end
+
+-- Build available choices
+local function menu()
+   local menu = {}
+   local out = outputs()
+   local choices = arrange(out)
+
+   for _, choice in pairs(choices) do
+      local cmd = "xrandr"
+      -- Enabled outputs
+      for i, o in pairs(choice) do
+         cmd = cmd .. " --output " .. o .. " --auto"
+         if i > 1 then
+            cmd = cmd .. " --right-of " .. choice[i-1]
+         end
+      end
+      -- Disabled outputs
+      for _, o in pairs(out) do
+         if not awful.util.table.hasitem(choice, o) then
+            cmd = cmd .. " --output " .. o .. " --off"
+         end
+      end
+
+      local label = ""
+      if #choice == 1 then
+         label = 'Only <span weight="bold">' .. choice[1] .. '</span>'
+      else
+         for i, o in pairs(choice) do
+            if i > 1 then label = label .. " + " end
+            label = label .. '<span weight="bold">' .. o .. '</span>'
+         end
+      end
+
+      menu[#menu + 1] = { label,
+                          cmd,
+                          icon_path}
+   end
+
+   return menu
+end
+
+-- Display xrandr notifications from choices
+local state = { iterator = nil,
+                timer = nil,
+                cid = nil }
+local function xrandr()
+   -- Stop any previous timer
+   if state.timer then
+      state.timer:stop()
+      state.timer = nil
+   end
+
+   -- Build the list of choices
+   if not state.iterator then
+      state.iterator = awful.util.table.iterate(menu(),
+                                                function() return true end)
```

Comment:
I don't think so. On third look, this code just iterates over integer entries. You can do something like that as follows easily (half-pseudocode follows)
```lua
if not state.index then
    state.menu = menu()
    state.index = 1
end
local next = state.menu[state.index]
state.index = state.index + 1
if not next then
[...]
```
The above would be easier, wouldn't it?